### ISSUE TITLE
onSelectedRowsChange have wrong typing

### ISSUE THREAD

**Issue by michaeljota** [jbetancur/react-data-table-component#509]
## Issue Check list
- [x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x] Read the README
- [x] You are using React 16.8.0+
- [x] You installed `styled-components`
- [ ] Include relevant code or preferably a [code sandbox](https://codesandbox.io/embed/react-data-table-sandbox-ccyuu
)

## Describe the bug

Currently this is being typed as:
```ts
  export interface IDataTableProps<T> {
    ...
    onSelectedRowsChange?: (selectedRowState: T) => void;
    ...
  }
```

## Expected behavior
Corrected typing (what is actually being send with the event):

```ts
  export interface IDataTableProps<T> {
    ...
    onSelectedRowsChange?: (selectedRowState: {
      allSelected: boolean;
      selectedCount: number;
      selectedRows: T[];
    }) => void;
    ...
  }
```

## Code Sandbox, Screenshots, or Relevant Code
Please include a codesandbox to help **expedite** troublshooting.

https://codesandbox.io/embed/react-data-table-sandbox-ccyuu

Otherwise, add screenshots and/or complete sample code to help explain your problem.

## Versions (please complete the following information)
 - React (RDT requires 16.8.0+)
 - Styled Components
 - Browser [e.g. chrome, safari, firefox]

**Comment by jbetancur** [comment]
WIll get this into a hot fix soon

**Comment by jbetancur** [comment]
Fixed in 6.3.6
